dragon/console_prompt.rb

```diff
@@ -63,11 +64,44 @@ def move_cursor_left
         update_cursor_position_px
       end
 
+      # ? Works well enough, could be better
+      # TODO: Maybe possibly add skipping multiple word-breaking characters
+      # TODO: (requires re-writing a lot of the code most likely)
+      # ?? how can we put in regex? the code would be cleaner that way \ amir please fix :P \\ abolish consoles :>
+      def move_cursor_left_word
+        str = @current_input_str[0..[@cursor_position - 1, 0].sort[1]]
+        # ? Can be changed, it was just taken from my editor settings :>
+        @cursor_position = ("`~!@#$%^&*-=+()[]{}\|;:'\",.<>/?_ \t\n\0".chars.map { |char|
+          (val = str.rindex char) ? (val + 1 == @cursor_position ? val : val + 1) : 0
+        }.sort.reverse[0])
+        update_cursor_position_px
+      end
+
       def move_cursor_right
         @cursor_position += 1 if @cursor_position < current_input_str.length
         update_cursor_position_px
       end
 
+      def move_cursor_right_word
+        str = @current_input_str[@cursor_position..@current_input_str.length]
+        # ? Can be changed, it was just taken from my editor settings :>
+        cand = ("`~!@#$%^&*-=+()[]{}\|;:'\",.<>/?_ \t\n\0".chars.map { |char|
+          (val = str.index char) ? val : 0
+        }.sort - [0])
+        (cand == []) ? @cursor_position = @current_input_str.length : @cursor_position += (cand[0]+0)
+        update_cursor_position_px
+      end
+
+      def move_cursor_home
+        @cursor_position = 0
+        update_cursor_position_px
```

Comment:
It's not directly related to your Pull Request....
But seeing `update_cursor_position_px` calls and checks to prevent the cursor_position from overflowing the line spread out everywhere through this class... 

it's maybe a good time now to extract a setter that takes care of all that..... and just call the setter everywhere....

Reply:
I agree but I had to work with what I was given (: